File: src/main/java/org/kite9/diagram/visualization/orthogonalization/AbstractOrthogonalizer.java
package org.kite9.diagram.visualization.orthogonalization;

import org.kite9.framework.logging.Kite9Log;
import org.kite9.framework.logging.Logable;

/**
 * Contains some utility methods that can be used by Orthogonalizers.
 * 
 * @author robmoffat
 * 
 */
public abstract class AbstractOrthogonalizer implements Orthogonalizer, Logable {
	
	protected Kite9Log log = new Kite9Log(this);

	public String getPrefix() {
		return "ORTH";
	}

	public boolean isLoggingEnabled() {
		return false;
	}
}

File: src/main/kotlin/org/kite9/diagram/visualization/orthogonalization/AbstractOrthogonalizer.kt
package org.kite9.diagram.visualization.orthogonalization

import org.kite9.diagram.visualization.orthogonalization.Orthogonalizer
import org.kite9.diagram.logging.Logable
import org.kite9.diagram.logging.Kite9Log

/**
 * Contains some utility methods that can be used by Orthogonalizers.
 *
 * @author robmoffat
 */
abstract class AbstractOrthogonalizer : Orthogonalizer, Logable {

    @JvmField
	protected var log = Kite9Log.instance(this)

    override val prefix: String
        get() = "ORTH"

    override val isLoggingEnabled: Boolean
        get() = false
}